### PR TITLE
feat: Remove secondary particles in Geant4 simulation

### PR DESCRIPTION
**Pull request by andiwand**
This allows us to ignore secondary particles in the Geant4 simulation which should speed things up and remove noise when unnecessary

### CODE REVIEW

File: Examples/Algorithms/Geant4/include/ActsExamples/Geant4/ParticleKillAction.hpp [hunk 1]
```diff
@@ -25,15 +25,16 @@ namespace ActsExamples {
 
 /// A G4SteppingAction that is called for every step in
 /// the simulation process.
-///
-/// It checks whether the particle can be killed according when its position
-/// exceeds the configured values for |z| or r.
 class ParticleKillAction : public G4UserSteppingAction {
```
Comment (paulgessinger) [inline Examples/Algorithms/Geant4/include/ActsExamples/Geant4/ParticleKillAction.hpp:28]: The name of the class is probably clear enough, I suppose?

File: Examples/Algorithms/Geant4/src/ParticleKillAction.cpp [hunk 2]
```diff
@@ -34,16 +34,20 @@ void ActsExamples::ParticleKillAction::UserSteppingAction(const G4Step* step) {
 
   const auto pos = convertLength * track->GetPosition();
   const auto time = convertTime * track->GetGlobalTime();
+  const bool secondary =
```
Comment (paulgessinger) [inline Examples/Algorithms/Geant4/src/ParticleKillAction.cpp:37]:
```suggestion
  const bool isSecondary =
```
?

File: Examples/Python/python/acts/examples/simulation.py [hunk 3]
```diff
@@ -606,6 +606,7 @@ def addGeant4(
     logLevel: Optional[acts.logging.Level] = None,
     killVolume: Optional[acts.Volume] = None,
     killAfterTime: float = float("inf"),
+    killSecondaries: float = False,
```
Comment (paulgessinger) [inline Examples/Python/python/acts/examples/simulation.py:609]:
```suggestion
    killSecondaries: bool = False,
```

File: Examples/Python/python/acts/examples/simulation.py [hunk 4]
```diff
@@ -630,8 +631,10 @@ def addGeant4(
         the output folder for the Root output, None triggers no output
     killVolume: acts.Volume, None
         if given, particles are killed when going outside of this volume.
-    killAfterTime: float, None
+    killAfterTime: float
         if given, particle are killed after the global time since event creation exceeds the given value
+    killSecondaries: float
```
Comment (paulgessinger) [inline Examples/Python/python/acts/examples/simulation.py:636]:
```suggestion
    killSecondaries: bool
```

**Review by paulgessinger — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results
### :white_check_mark: All tests successful

|status|job|
|------|---|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30795872) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30795871) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30795870) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30795869) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsTrkSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30795868) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30795867) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30795866) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30795865) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30795864) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30795863) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30795862) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30795861) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30795860) |
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30795859) |
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30795858) |
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30795857) |
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30795856) |

**Pull request merged**